### ISSUE TITLE
[css-inline] Arabic drop-caps

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#2399]
@Behnam provided the necessary info in https://github.com/w3c/csswg-drafts/issues/698#issuecomment-297472304 to define shaping behavior, so we should spec it.

**Comment by r12a** [comment]
alreq TF notified

**Comment by r12a** [comment]
> The few examples I've seen of traditional practice (see #698 (comment) and the linked discussion) don't appear to support shaping across more general drop-cap-like formatting: the large dropped, boxed initial in https://upload.wikimedia.org/wikipedia/commons/6/6a/Hafezeshamlu02.jpg, for example, is not shaped in an initial form.

Do we have a source which we believe is reliable that _does_ show enlarged initial letters using joining forms? @behnam, @shervinafshar, @haledhosny, @sahafshar, @ntounsi, @mostafa, any ideas?

**Comment by shervinafshar** [comment]
@r12a, this topic has been bouncing around for a bit now. My personal take is that since this is a novelty of a typographical practice in Arabic script text, there is no reliable source. However, the research shows that (a) most cases of drop-caps in Arabic script texts are not using joining forms; (b) through my research on the matter, I believe that there is also some precedence for the drop-caps in Arabic script text to _use of joining forms_. This requires further investigation, but I reproduce the case that I'm referring to as a rare precedence.

What follows is from the volume [_Compendium of Latin Translations of Persian Astronomical Tables_](https://www.wdl.org/en/item/10676/#institution=qatar-national-library), selected portions of [_Zīj-i Sultānī_](https://en.wikipedia.org/wiki/Zij-i_Sultani) translated into Latin and published by Oxford University Press in 1655.

![image](https://user-images.githubusercontent.com/875962/49882078-ad7fba00-fde4-11e8-8487-b82f50932f65.png)

![image](https://user-images.githubusercontent.com/875962/49881977-6d203c00-fde4-11e8-88d2-796168ac9007.png)

Further pointers:
– [my thread on Persian Computing](https://groups.google.com/forum/#!topic/persian-computing/W-Upl6DAEcg)
– [Liam Quin's blogpost](https://barefootliam.blogspot.com/2014/05/drop-caps-other-writing-systems-other.html) has a section with his findings, points to the thread above

**Comment by sahafshar** [comment]
I’ve only ever seen examples of this form as non-connected (from what I recall it’s a very rare practice in general, and I can’t think of many modern day uses of it. Attached example is from Ketab’e Hafte (Book of the Week) published in Tehran, 1962. 
I agree with Shervin that it’s unlikely any reliable resource can be found on this, as it’s a purely typographic convention and not something–I reckon– you would ever see in calligraphy or manuscripts. 


 
--

Sahar Afshar

PhD Candidate | Research Assistant 
Centre for Printing History & Culture
Faculty of Art, Design and Media, Birmingham City University
Parkside Building, 5 Cardigan Street, Birmingham B4 7BD

www.cphc.org.uk  @sahafshar

**Comment by r12a** [comment]
I wonder whether the answer here is to assume that by default arabic versals are _not_ joining forms, but allow authors to turn them into joining forms using ZWJ if desired??

In that case, would we need a special rule to say that ZWJ is kept with the preceding letter when using `::first-letter` selection?

(Btw, @sahafshar your image didn't seem to make it into the thread.)

**Comment by r12a** [comment]
This is a test for the new i18n notification tooling. Please ignore. I'll delete this comment later.

**Comment by r12a** [comment]
This is a test for the new i18n notification tooling. Please ignore. I'll delete this comment later.

**Comment by r12a** [comment]
Fwiw (perhaps not much) if you open [this test](https://w3c.github.io/i18n-tests/exploratory/initials/int-initial-letters?lang=ar&fontSize=69&width=380&webkitInitialLetters=2&text=%D9%8A%E2%80%8D%D8%AD%D9%82%20%D9%84%D9%83%D9%84%20%D9%81%D8%B1%D8%AF%20%D8%A3%D9%86%20%D9%8A%D8%BA%D8%A7%D8%AF%D8%B1%20%D8%A3%D9%8A%D8%A9%20%D8%A8%D9%84%D8%A7%D8%AF%20%D8%A8%D9%85%D8%A7%20%D9%81%D9%8A%20%D8%B0%D9%84%D9%83%20%D8%A8%D9%84%D8%AF%D9%87%20%D9%83%D9%85%D8%A7%20%D9%8A%D8%AD%D9%82%20%D9%84%D9%87%20%D8%A7%D9%84%D8%B9%D9%88%D8%AF%D8%A9%20%D8%A5%D9%84%D9%8A%D9%87.) in Safari (which is the only major browser i know of that supports initial-letter behaviour) you'll see that when selecting the first letter of a word where i had inserted ZWJ after the initial letter, ::first-letter automatically picked up the ZWJ and rendered the enlarged initial as a joining form.

If there is no ZWJ (see [this test](https://w3c.github.io/i18n-tests/exploratory/initials/int-initial-letters?lang=ar&fontSize=69&width=380&webkitInitialLetters=2&text=%D9%8A%D8%AD%D9%82%20%D9%84%D9%83%D9%84%20%D9%81%D8%B1%D8%AF%20%D8%A3%D9%86%20%D9%8A%D8%BA%D8%A7%D8%AF%D8%B1%20%D8%A3%D9%8A%D8%A9%20%D8%A8%D9%84%D8%A7%D8%AF%20%D8%A8%D9%85%D8%A7%20%D9%81%D9%8A%20%D8%B0%D9%84%D9%83%20%D8%A8%D9%84%D8%AF%D9%87%20%D9%83%D9%85%D8%A7%20%D9%8A%D8%AD%D9%82%20%D9%84%D9%87%20%D8%A7%D9%84%D8%B9%D9%88%D8%AF%D8%A9%20%D8%A5%D9%84%D9%8A%D9%87.)) then the initial letter is unjoined, but the 2nd letter in the word is joined.

**Comment by svgeesus** [comment]
@faceless2 a nice drop cap testcase for you

**Comment by fantasai** [comment]
I'm inclined to spec the behavior @r12a describes in https://github.com/w3c/csswg-drafts/issues/2399#issuecomment-577520633 Happy to do something different if there's some clear consensus as to what's correct, though. :)

**Comment by jfkthame** [comment]
> If there is no ZWJ (see [this test](https://w3c.github.io/i18n-tests/exploratory/initials/int-initial-letters?lang=ar&fontSize=69&width=380&webkitInitialLetters=2&text=%D9%8A%D8%AD%D9%82%20%D9%84%D9%83%D9%84%20%D9%81%D8%B1%D8%AF%20%D8%A3%D9%86%20%D9%8A%D8%BA%D8%A7%D8%AF%D8%B1%20%D8%A3%D9%8A%D8%A9%20%D8%A8%D9%84%D8%A7%D8%AF%20%D8%A8%D9%85%D8%A7%20%D9%81%D9%8A%20%D8%B0%D9%84%D9%83%20%D8%A8%D9%84%D8%AF%D9%87%20%D9%83%D9%85%D8%A7%20%D9%8A%D8%AD%D9%82%20%D9%84%D9%87%20%D8%A7%D9%84%D8%B9%D9%88%D8%AF%D8%A9%20%D8%A5%D9%84%D9%8A%D9%87.)) then the initial letter is unjoined, but the 2nd letter in the word is joined.

@r12a When you say "but the 2nd letter in the word is joined", do you mean it takes a right-joining (medial or final) form, as if it were joined to the initial letter? Or just that it is joined to what follows it?

(As far as I can see when testing here, it takes an initial form; i.e. it does not shape as though joined to the initial letter.)

**Comment by fantasai** [comment]
Got an iOS screenshot I got from my friend... so it does look like Safari is breaking the connection on both sides.

That said, using the correct connecting form between the second part of the word and the initial letter seems to me would be more readable than actually breaking this connection. Similar concept to how Latin drop-caps kern in the rest of the first word into the drop-cap, to help maintain a clearer connection between the first letter and the rest of its word.

I also understand that using isolated form for a drop-cap just looks a lot better; that's probably why it's more common per @shervinafshar and @sahafshar’s comments.

So basically we have three options here:
- Break the connection on both sides: isolated form drop-cap, initial form rest of the word.
- Connect both sides: Initial form drop-cap, connected (medial) form rest of the word.
- Split model: Isolate drop-cap, connected (medial) form rest of the word.

I think the third option is actually the best one. Curious to hear what native users of the writing system think.

Here's a mockup of all three options, for reference:
![arabic-drop-cap](https://user-images.githubusercontent.com/725717/83198247-a7be8700-a0f3-11ea-9aa5-72f78e021d53.png)

**Comment by sahafshar** [comment]
I agree @fantasai, the third option makes most sense to me. Sorry @r12a you're right, I forgot the reference picture in my original response on the thread. I've attached it here. As you can see, it also follows the same logic as the split model. 
![Dropcaps](https://user-images.githubusercontent.com/17496291/83198603-6e0f6f80-a137-11ea-82be-268a2266381a.jpg)

**Comment by fantasai** [comment]
Tentatively updated the editor's draft with text for this behavior given @sahafshar’s confirmation. I'll wait on more feedback before closing the issue, though.

**Comment by faceless2** [comment]
This change isn't script-specific? Seems reasonable. Given how hard it was finding examples for arabic etc. I'm pretty sure we'd be basing a decision on a sample size of zero for N'Ko.

**Comment by fantasai** [comment]
@faceless2 I think in the absence of conclusive information to the contrary, it should apply to all scripts. :) The principles that make it the most sensible option (see https://github.com/w3c/csswg-drafts/issues/2399#issuecomment-635630662) are generally-applicable.
